uv_work_t and uv_after_work_cb: compatibility with node >= 0.9.4

This is the issue:

```
node-gyp build configure
gyp info it worked if it ends with ok
gyp info using node-gyp@0.10.9
gyp info using node@0.10.15 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory `/home/pablo/fic/portaudio-node/repo/build'
  CXX(target) Release/obj.target/portAudio/src/nodePortAudio.o
../src/nodePortAudio.cpp: In function ‘int nodePortAudioCallback(const void*, void*, long unsigned int, const PaStreamCallbackTimeInfo*, PaStreamCallbackFlags, void*)’:
../src/nodePortAudio.cpp:352:84: error: invalid conversion from ‘void (*)(uv_work_t*) {aka void (*)(uv_work_s*)}’ to ‘uv_after_work_cb {aka void (*)(uv_work_s*, int)}’ [-fpermissive]
In file included from /home/pablo/.node-gyp/0.10.15/src/node.h:61:0,
                 from ../src/nodePortAudio.h:2,
                 from ../src/nodePortAudio.cpp:2:
/home/pablo/.node-gyp/0.10.15/deps/uv/include/uv.h:1432:15: error:   initializing argument 4 of ‘int uv_queue_work(uv_loop_t*, uv_work_t*, uv_work_cb, uv_after_work_cb)’ [-fpermissive]
```

There's been some changes in Node since 0.8
Here's the code for compatibility with newer node apis

https://github.com/joyent/node/wiki/Api-changes-between-v0.8-and-v0.10

The uv_after_work_cb signature has changed to take a second integer argument indicating status. For backwards compatibility, explicitly cast the 4th argument to uv_queue_work

This is the example:
https://github.com/rbranson/node-ffi/commit/fdeff41ae8b1cca31d4707d7b61253c45181b8fa

I think that changing the call like there fixes it

regards

--- a/src/nodePortAudio.cpp
+++ b/src/nodePortAudio.cpp
@@ -349,7 +349,7 @@ static int nodePortAudioCallback(
     if(data->readIdx == data->writeIdx) {
       uv_work_t* req = new uv_work_t();
       req->data = data;
-      uv_queue_work(uv_default_loop(), req, EIO_EmitUnderrun, EIO_EmitUnderrunAfter);
+      uv_queue_work(uv_default_loop(), req, EIO_EmitUnderrun, (uv_after_work_cb) EIO_EmitUnderrunAfter);
       return paContinue;
     }
     *out++ = data->buffer[data->readIdx++];

--- a/src/nodePortAudio.h
+++ b/src/nodePortAudio.h
@@ -2,6 +2,7 @@
 #include <node.h>
 #include <v8.h>
 #include <node_buffer.h>
+#include <cstring>
 
 v8::Handle<v8::Value> Open(const v8::Arguments& args);
 v8::Handle<v8::Value> GetDevices(const v8::Arguments& args);

